pkg/connections/connections.go

```diff
@@ -0,0 +1,331 @@
+	err := c.List(context.TODO(), nodeList)
+	if err != nil {
+		return nil, err
+	}
+
+	nodesRolesMap := map[string][]string{}
+	for _, node := range nodeList.Items {
+		nodesRolesMap[node.Name] = types.GetNodeRoles(&node)
+	}
+
+	return &ConnectionCheck{
+		c,
+		podUtils,
+		customEntriesPath,
+		customEntriesFormat,
+		format,
+		destDir,
+		env,
+		deployment,
+		nodesRolesMap,
+	}, nil
+}
+
+func (cc *ConnectionCheck) GenerateSS() (*types.ComMatrix, []byte, []byte, error) {
+	nodesList := &corev1.NodeList{}
+	err := cc.List(context.TODO(), nodesList)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	nodesComDetails := []types.ComDetails{}
+	err = cc.podUtils.CreateNamespace(consts.DefaultDebugNamespace)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	defer func() {
+		err := cc.podUtils.DeleteNamespace(consts.DefaultDebugNamespace)
+		if err != nil {
+			fmt.Printf("failed to delete debug namespace: %v", err)
+		}
+	}()
+	var ssOutTCP, ssOutUDP []byte
+	nLock := &sync.Mutex{}
+	g := new(errgroup.Group)
+	for _, n := range nodesList.Items {
+		node := n.DeepCopy()
+		g.Go(func() error {
+			debugPod, err := cc.podUtils.CreatePodOnNode(node.Name, consts.DefaultDebugNamespace, consts.DefaultDebugPodImage)
+			if err != nil {
+				return err
+			}
+			defer func() {
+				err := cc.podUtils.DeletePod(debugPod)
+				if err != nil {
+					fmt.Printf("failed cleaning debug pod %s: %v", debugPod, err)
+				}
+			}()
+
+			cds, ssTCP, ssUDP, err := cc.createSSOutputFromNode(debugPod, node)
+			if err != nil {
+				return err
+			}
+			nLock.Lock()
+			defer nLock.Unlock()
+			ssTCPLine := fmt.Sprintf("node: %s\n%s\n", node.Name, string(ssTCP))
+			ssUDPLine := fmt.Sprintf("node: %s\n%s\n", node.Name, string(ssUDP))
+
+			nodesComDetails = append(nodesComDetails, cds...)
+			ssOutTCP = append(ssOutTCP, []byte(ssTCPLine)...)
+			ssOutUDP = append(ssOutUDP, []byte(ssUDPLine)...)
+			return nil
+		})
+	}
+
+	err = g.Wait()
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	cleanedComDetails := types.CleanComDetails(nodesComDetails)
+	ssComMat := types.ComMatrix{Matrix: cleanedComDetails}
+
+	return &ssComMat, ssOutTCP, ssOutUDP, nil
+}
+
+func (cc *ConnectionCheck) WriteSSRawFiles(ssOutTCP, ssOutUDP []byte) error {
+	tcpFile, err := os.OpenFile(path.Join(cc.destDir, "raw-ss-tcp"), os.O_TRUNC|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		return err
+	}
+	defer tcpFile.Close()
+
+	udpFile, err := os.OpenFile(path.Join(cc.destDir, "raw-ss-udp"), os.O_TRUNC|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		return err
+	}
+	defer udpFile.Close()
+
+	_, err = tcpFile.Write(ssOutTCP)
+	if err != nil {
+		return fmt.Errorf("failed writing to file: %s", err)
+	}
+
+	_, err = udpFile.Write(ssOutUDP)
+	if err != nil {
+		return fmt.Errorf("failed writing to file: %s", err)
+	}
+
+	return nil
+}
+
+func (cc *ConnectionCheck) createSSOutputFromNode(debugPod *corev1.Pod, node *corev1.Node) ([]types.ComDetails, []byte, []byte, error) {
```

Comment:
u can send node name and not the  node *corev1.Node
since we are just using name